lib/group-update.js

```diff
@@ -0,0 +1,335 @@
+const ERR = require('async-stacktrace');
+const util = require('util');
+const _ = require('lodash');
+const streamifier = require('streamifier');
+const csvtojson = require('csvtojson');
+
+const logger = require('./logger');
+const serverJobs = require('./server-jobs');
+const sqldb = require('@prairielearn/prairielib/sql-db');
+const sqlLoader = require('@prairielearn/prairielib/sql-loader');
+
+const sql = sqlLoader.loadSqlEquiv(__filename);
+
+module.exports = {
+
+    /**
+     * Update question instance scores from a CSV file.
+     *
+     * @param {number} assessment_id - The assessment to update.
+     * @param {csvFile} csvFile - An object with keys {originalname, size, buffer}.
+     * @param {number} user_id - The current user performing the update.
+     * @param {number} authn_user_id - The current authenticated user.
+     * @param {function} callback - A callback(err, job_sequence_id) function.
+     */
+    uploadInstanceGroups(assessment_id, csvFile, user_id, authn_user_id, callback) {
```

Comment:
There doesn't seem to be any locking here. What happens if two instructors try and modify groups at the same time? Unless the code is handling this case already, it might be easiest to use `lib/named-locks` to get a lock on the group configuration for this assessment_id.